src/extension.ts

```diff
@@ -237,8 +237,15 @@ export async function activate(ctx: vscode.ExtensionContext) {
     if (configureOnOpen) {
       let cancel = false;
       if (!configurationChosen && mesonFiles.length > 1) {
-        const items = mesonFiles.map((file, index) => ({ index: index, label: relative(root, file.fsPath) }));
-        items.sort((a, b) => a.label.localeCompare(b.label));
+        const itemsBase = mesonFiles.map((file, index) => ({ index: index, label: relative(root, file.fsPath) }));
+        const items = itemsBase.map((item) => ({ ...item, depth: item.label.split(sep).length }));
```

Comment:
Can this be done more concisely? I don't understand why it is O(2n) instead of O(n)

Reply:
I don't know a way to access `label` without recomputing `relative` in the first `map` call.

Reply:
```typescript
const items = mesonFiles.map((file, index) => {
  const label = relative(root, file.fsPath);
  const depth = label.split(sep).length;

  return { index, label, depth };
});
```

Reply:
Thanks, done.